challenge_scoring/metrics/scoring.py

```diff
@@ -141,101 +152,122 @@ def score_submission(streamlines_fname,
-    scores['f1_score_per_bundle'] = {k: v["f1_score"] for k, v in found_vbs_info.items()}
+        save_valid_connections(found_vbs_info, sft, segmented_out_dir,
+                               segmented_base_name, out_tract_type,
+                               save_vbs=save_VBs,
+                               save_full_vc=save_full_vc)
+
+    candidate_ic_strl_indices = np.setdiff1d(range(total_strl_count),
+                                             VC_indices)
+    if compute_ic_ib:
+        logging.info("Starting IC, IB scoring")
+
+        candidate_ic_indices = []
+        rejected_indices = []
+
+        # Chosen from GT dataset
+        length_thres = 35.
+
+        # Filter streamlines that are too short, consider them as NC
+        for idx in candidate_ic_strl_indices:
+            if slength(sft.streamlines[idx]) >= length_thres:
+                candidate_ic_indices.append(idx)
+            else:
+                rejected_indices.append(idx)
+
+        logging.debug('Found {} candidate IC'
+                      .format(len(candidate_ic_indices)))
+        logging.debug('Found {} streamlines that were too short'
+                      .format(len(rejected_indices)))
+
+        ic_counts = 0
+        nb_ib = 0
+
+        if len(candidate_ic_indices):
+            additional_rejected_indices, ic_counts, nb_ib = \
+                group_and_assign_ibs(sft, candidate_ic_indices,  ROIs,
+                                     save_IBs, save_full_ic, segmented_out_dir,
+                                     segmented_base_name, ref_anat_fname,
+                                     out_tract_type)
+
+            rejected_indices.extend(additional_rejected_indices)
+
+        if ic_counts != len(candidate_ic_strl_indices) - len(rejected_indices):
+            raise ValueError("Some streamlines were not correctly assigned to "
+                             "NC")
+
+        IC = len(candidate_ic_strl_indices) - len(rejected_indices)
+    else:
+        IC = 0
+        nb_ib = 0
+        rejected_indices = candidate_ic_strl_indices
+
+    if len(rejected_indices) > 0 and save_full_nc:
+        out_nc_fname = os.path.join(
+            segmented_out_dir,
+            '{}_NC.{}'.format(segmented_base_name, out_tract_type))
+
+        save_tractogram(sft[rejected_indices], out_nc_fname, bbox_valid_check=False)
+
+    logging.debug("Preparing summary of results")
+
+    nb_VB_found = [v['nb_streamlines'] > 0 for k,
+                   v in found_vbs_info.items()].count(True)
+    streamlines_per_bundle = {
+        k: v['nb_streamlines']
+        for k, v in found_vbs_info.items() if v['nb_streamlines'] > 0}
+
+    # Converting np.float to floats for json dumps
+    NC = len(rejected_indices)
+    scores = {'version': 2,
+              'algo_version': 5,
+              'VC': VC,
+              'VC_pct': VC / total_strl_count,
+              'IC': IC,
+              'IC_pct': IC / total_strl_count,
```

Comment:
Just realized this was added when testing it (or I may have forgot). Should we keep this @EmmaRenauld ? This might induce questions or break stuff for people already using the tool, as "IC" was previously in percentages. Maybe switch it so that VC/IC/NC is still in percentages and add "VC/IC/NC_count" ?

Reply:
Ok for me, no problem.

Reply:
Done in [3528a8f](https://github.com/scilus/ismrm_2015_tractography_challenge_scoring/pull/6/commits/3528a8f73331572bc9ed88801eab589b1846e0ac)